caffe-grid/src/main/python/examples/README-coco.md

```diff
@@ -0,0 +1,121 @@
+        --conf spark.task.cpus=${CORES_PER_WORKER} \
+    	--conf spark.driver.extraLibraryPath="${DYLD_LIBRARY_PATH}" \
+	--conf spark.executorEnv.DYLD_LIBRARY_PATH="${DYLD_LIBRARY_PATH}" \
+        --class com.yahoo.ml.caffe.CaffeOnSpark  \
+	${CAFFE_ON_SPARK}/caffe-grid/target/caffe-grid-0.1-SNAPSHOT-jar-with-dependencies.jar \
+            -train \
+            -conf bvlc_reference_solver.prototxt \
+            -model /tmp/coco/bvlc_reference_caffenet.caffemodel \
+            -devices 1
+    hadoop fs -ls /tmp/coco/bvlc_reference_caffenet.caffemodel
+    popd
+##### (5) Train the lstm
+    pushd ${CAFFE_ON_SPARK}/data/
+    spark-submit --master ${MASTER_URL} \
+        --files lrcn_cos.prototxt,lrcn_solver.prototxt \
+	--conf spark.cores.max=${TOTAL_CORES} \
+        --conf spark.task.cpus=${CORES_PER_WORKER} \
+        --conf spark.driver.extraLibraryPath="${DYLD_LIBRARY_PATH}" \
+        --conf spark.executorEnv.DYLD_LIBRARY_PATH="${DYLD_LIBRARY_PATH}" \       
+        --class com.yahoo.ml.caffe.CaffeOnSpark  \
+  	  ${CAFFE_ON_SPARK}/caffe-grid/target/caffe-grid-0.1-SNAPSHOT-jar-with-dependencies.jar \
+	     -train \
+             -conf lrcn_solver.prototxt \
+             -devices 1 \
+             -resize \
+             -weights /tmp/coco/bvlc_reference_caffenet.caffemodel \
+             -model /tmp/coco/parquet/lrcn_coco.model
+    popd
+
+##### (6) Submit the data for inference
+    Note that the below files also need to be shipped as shown
+    #-model the image-lstm pretrained model to ship 
+    #-imagenet the image network definition
+    #-lstmnet the lstm network definition
+    #-vocab the vocabulary file (produced from above) for the given train set
+    #-input the input embedding produced above
+    #-output the path where to write the desired output
+
+    pushd ${CAFFE_ON_SPARK}/data/
+    ln -s ~/Python2.7.10 Python2.7.10
+    unzip ${CAFFE_ON_SPARK}/caffe-grid/target/caffeonsparkpythonapi.zip
+    cat /tmp/coco/parquet/vocab/part* > vocab.txt
+    rm -rf /tmp/coco/parquet/df_caption_results_train2014
+    spark-submit --master ${MASTER_URL} \
+    		 --conf spark.cores.max=${TOTAL_CORES} \
+    		 --conf spark.task.cpus=${CORES_PER_WORKER} \    
+    		 --conf spark.driver.extraLibraryPath="${DYLD_LIBRARY_PATH}:Python2.7.10/lib" \
+    		 --conf spark.executorEnv.LD_LIBRARY_PATH="${DYLD_LIBRARY_PATH}:Python2.7.10/lib" \
+    		 --conf spark.pythonargs="-model /tmp/coco/parquet/lrcn_coco.model -imagenet lstm_deploy.prototxt -lstmnet lrcn_word_to_preds.deploy.prototxt -vocab vocab.txt -input /tmp/coco/parquet/df_embedded_train2014 -output /tmp/coco/parquet/df_caption_results_train2014" examples/ImageCaption.py
+    popd
+##### (7) Launch IPython Notebook
+    export IPYTHON_OPTS="notebook --no-browser --ip=127.0.0.1"
+    pushd ${CAFFE_ON_SPARK}/data/
+    ln -s ~/Python2.7.10 Python2.7.10
+    unzip ${CAFFE_ON_SPARK}/caffe-grid/target/caffeonsparkpythonapi.zip
+    cat /tmp/coco/parquet/vocab/part* > vocab.txt
```

Comment:
We should not need the above command:  cat /tmp/coco/parquet/vocab/part* > vocab.txt